nipype/interfaces/afni/preprocess.py

```diff
@@ -2556,6 +2556,181 @@ def _format_arg(self, name, trait_spec, value):
+             "WB_Z_ROI_001+orig. "
+             "With this option, one can replace the int (such as '001') "
+             "with the string label (such as 'L-thalamus') "
+             "*if* one has a labeltable attached to the file",
+        argstr="-ts_wb_strlabel")
+    nifti = traits.Bool(
+        desc="output any correlation map files as NIFTI files "
+             "(default is BRIK/HEAD). Only useful if using "
+             "'-ts_wb_corr' and/or '-ts_wb_Z'",
+        argstr="-nifti")
+    output_mask_nonnull = traits.Bool(
+        desc="internally, this program checks for where there are "
+             "nonnull time series, because we don't like those, in "
+             "general.  With this flag, the user can output the "
+             "determined mask of non-null time series.",
+        argstr="-output_mask_nonnull")
+    push_thru_many_zeros = traits.Bool(
+        desc="by default, this program will grind to a halt and "
+             "refuse to calculate if any ROI contains >10 percent "
+             "of voxels with null times series (i.e., each point is "
+             "0), as of April, 2017.  This is because it seems most "
+             "likely that hidden badness is responsible. However, "
+             "if the user still wants to carry on the calculation "
+             "anyways, then this option will allow one to push on "
+             "through.  However, if any ROI *only* has null time "
+             "series, then the program will not calculate and the "
+             "user will really, really, really need to address their masking",
+        argstr="-push_thru_many_zeros")
+    ignore_LT = traits.Bool(
+        desc="switch to ignore any label table labels in the "
+             "'-in_rois' file, if there are any labels attached",
+        argstr="-ignore_LT")
+    out_file = File(
+        desc="output file name part",
+        name_template="%s_netcorr",
+        argstr="-prefix %s",
+        position=1,
+        name_source="in_file",
+    )
+
+class NetCorrOutputSpec(TraitedSpec):
+    out_matrix = File(desc="output text file for correlation stats")
+
+class NetCorr(AFNICommand):
+    """Calculate correlation matrix of a set of ROIs (using mean time series of
+    each). Several networks may be analyzed simultaneously, one per brick.
+
+    For complete details, see the `3dNetCorr Documentation
+    <https://afni.nimh.nih.gov/pub/dist/doc/program_help/3dNetCorr.html>`_.
+
+    Examples
+    --------
+    >>> from nipype.interfaces import afni
+    >>> ncorr = afni.NetCorr()
+    >>> ncorr.inputs.in_file = 'functional.nii'
+    >>> ncorr.inputs.mask = 'mask.nii'
+    >>> ncorr.inputs.in_rois = 'maps.nii'
+    >>> ncorr.inputs.ts_wb_corr = True
+    >>> ncorr.inputs.ts_wb_Z = True
+    >>> ncorr.inputs.fish_z = True
+    >>> ncorr.inputs.out_file = 'sub0.tp1.ncorr'
+    >>> ncorr.cmdline
+    '3dNetCorr -prefix sub0.tp1.ncorr -inset functional.nii -mask mask.nii -in_rois maps.nii -ts_wb_corr -ts_wb_Z -fish_z'
```

Comment:
For the things without explicit positions, the arguments get sorted, IIRC. In any event, I think you want:

```suggestion
    '3dNetCorr -prefix sub0.tp1.ncorr -fish_z -inset functional.nii -in_rois maps.nii -mask mask.nii -ts_wb_Z -ts_wb_corr'
```

Reply:
@effigies: Thanks. I applied your suggestion. How the arguments are sorted?

Reply:
https://github.com/nipy/nipype/blob/47fe00b387e205f3f884691a208ef7f20df4119f/nipype/interfaces/base/core.py#L967-L1008

And the heart is:

https://github.com/nipy/nipype/blob/47fe00b387e205f3f884691a208ef7f20df4119f/nipype/interfaces/base/core.py#L983